### ISSUE TITLE
Detect and delete duplicate files and change duplicate to "reposted from" refering to original poster

### ISSUE THREAD

**Issue by spamcheck** [souperio/issues#3]

**Comment by v2px** [comment]
I-I don't get what this means, isn't this how it works now?

**Comment by spamcheck** [comment]
I like that there is no duplicate reposts on /everyone and /friends but i'm talking about duplicate files.
Soup apparently had a feature that let you check who's the original poster of image so if you "stole" it from someone you could still see who really posted it first by adding something into URL.
I was thinking if they were able to do such black magic (probably domain image reverse search) maybe we could go further and just check if the image wasn't uploaded by somebody else before and force it to be marked as "reposted from"(in case it was).
I noticed some people like to "steal" other peoples memes instead of just reposting them and some people just have same sources of content as others and post the same content unconsciously.
This would help avoid multiple posting of the same image/file and loosen the servers but i'm not sure if it's doable.

**Comment by v2px** [comment]
Aha! I'm happy to say that this is already implemented, duplicate uploads all link to the same physical file:
> This would help avoid multiple posting of the same image/file and loosen the servers but i'm not sure if it's doable.

This however is not:
> and force it to be marked as "reposted from"

As reposts are deleted when the original post is deleted I'm not sure if this would be a feature or a misfeature tho.

**Comment by spamcheck** [comment]
Wow nice.
> As reposts are deleted when the original post is deleted
Random posts disapering from my soup doesn't sound good.
Also i used to post some memes I didn't want to keep on my soup so I jest let few people repost it, then deleted it.

**Comment by v2px** [comment]
> Random posts disappearing from my soup doesn't sound good.
not this issue but I'll say it here anyway: sadly this is the only solution that plays well with EU laws. so after thinking about this hard, this will stay in. posts of deleted users will be deleted with them. if anyone has a good idea how to handle this better please make a new issue about it.

now about this issue:
re-uploads are detected by file hash and stored only once. reposts without linking to the original are a feature. if this doesn't fit well with your OCD I could adding a settings switch that will warn you about this. how does that sound?

**Comment by v2px** [comment]
didn't mean to close smh
